pkg/storage/azureblob/azureblob.go

```diff
@@ -0,0 +1,155 @@
+package azureblob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+)
+
+type client interface {
+	UploadWithContext(ctx context.Context, path, contentType string, content io.Reader) error
+	BlobExists(ctx context.Context, path string) (bool, error)
+	ReadBlob(ctx context.Context, path string) (io.ReadCloser, error)
+	ListBlobs(ctx context.Context, prefix string) ([]string, error)
+	DeleteBlob(ctx context.Context, path string) error
+	GetTimeout() time.Duration
+}
+
+type azureBlobStoreClient struct {
+	containerURL *azblob.ContainerURL
+	timeout      time.Duration
+}
+
+func newBlobStoreClient(accountURL *url.URL, accountName, accountKey, containerName string, timeout time.Duration) (*azureBlobStoreClient, error) {
+	const op errors.Op = "storage.azure.azureblob"
```

Comment:
and ops constants in this file should be as well `azureblob.{methodName}` 
in this case `azureblob.newBlobStoreClient` 
for List method
`azureblob.List`

Reply:
This was a dumb mistake 😦 , hope I don't do any new blunders 😅